### PR TITLE
Update docker to support filtered data download

### PR DESCRIPTION
**Pull request by dianashk**
Connected to pelias/whosonfirst#243

### Overview
The docker script was updated to support downloading the data using the new filter mechanism in the pelias.json config file. A sample pelias.json was added to make the docker setup easier. Documenation was updated to include docker setup, which wasn't mentioned in there at all before.

### CODE REVIEW

**Comment by tigerlily-he** [comment]
Docker installation works on OSX 10.12.5, Docker version 17.06.0-ce. Very quick and convenient!

File: Dockerfile [hunk 1]
```diff
@@ -8,13 +8,34 @@ RUN curl -sL https://deb.nodesource.com/setup_4.x | bash -
...
+ENV PELIAS_CONFIG '/code/pelias.json'
+
+# clone repo
+RUN git clone https://github.com/pelias/whosonfirst.git /code/pelias/whosonfirst
+
+# change working dir
+WORKDIR /code/pelias/whosonfirst
+
+# install npm dependencies
+RUN git checkout filtered-download
```
Comment (tigerlily-he) [inline Dockerfile]: This won't be needed once the changes from filtered-download branch is merged with master?

File: Dockerfile [hunk 2]
```diff
@@ -8,13 +8,33 @@ RUN curl -sL https://deb.nodesource.com/setup_4.x | bash -
...
+
+# clone repo
+RUN git clone https://github.com/pelias/whosonfirst.git /code/pelias/whosonfirst
+
+# change working dir
+WORKDIR /code/pelias/whosonfirst
+
+# install npm dependencies
 RUN npm install
+RUN npm run download
```
Comment (missinglink) [inline Dockerfile]: are you sure you want to run this at `build time`? that will mean all the WOF data files will be stored inside the image.

I think it's a better idea if docker **images** only contain code and that the user executes scripts inside a **container** at `run time` to download the data on to an ephemeral/persistent disk of choice.

having a clearer separation of code from data will allow for much more flexibility and will also reduce the size of images and the frequency they need to be rebuilt.

also worth considering that the filtered download options in `pelias.json` will only download a subset of the global data, making any changes or updates to this data would require a new **image** be built, even if the code remained the same.

File: README.md [hunk 3]
```diff
@@ -17,6 +17,33 @@ $ cd pip-service
...
+There is an api key placeholder in there as well, that should be replaced with
+the user's [Mapzen Developer](mapzen.com/developers) api key.
+
+```bash
+$ git clone git@github.com:pelias/pip-service.git
+$ cd pip-service
+docker build -t pelias/pip .
+docker run -d -p 3102:3102 --name pip1 pelias/pip
+```
+
```
Comment (missinglink) [inline README.md]: you could achieve this by adding one more command here (after the `docker build` step), such as:

```bash
docker run --rm -it pelias/pip bach -c 'cd /code/pelias/whosonfirst && npm run download'
```

changes to `pelias.json` could be made with a bind-mount such as:

```bash
docker run -v /home/foo/pelias.json:/code/pelias.json
```

File: docker/pelias.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "imports": {
+    "whosonfirst": {
+      "datapath": "/whosonfirst",
+      "importPostalcodes": true,
+      "importVenues": false,
+      "importPlace": "101715829",
+      "api_key": "your-api-key"
+    }
+  }
+}
```
Comment (missinglink) [inline docker/pelias.json]: this example `pelias.json` seems to be tied to a specific task (101715829) without any mention of why or how to change it.

**Comment by dianashk** [comment]
These changes are no longer relevant.

**Pull request closed without merging**